src/python/pants/backend/python/tasks2/setup_py.py

```diff
@@ -607,10 +611,10 @@ def create(target):
           self.context.log.info('Writing {}'.format(sdist_path))
           shutil.move(setup_runner.sdist(), sdist_path)
           safe_rmtree(setup_dir)
-          executed[target] = sdist_path
+          python_dists.add(target, os.path.dirname(sdist_path)).append(os.path.basename(sdist_path))
```

Comment:
The 'basedir' is generally expected to be a "package root" or something similar. There's no such concept here, so you're effectively having to use the file's immediately enclosing dir as a fake basedir.  This is why it's better to use "data products", where you can pick your own data structure, and not worry about this. See above.